code/modules/integrated_electronics/subtypes/input.dm

```diff
@@ -524,28 +524,36 @@
 	var/datum/integrated_io/I = inputs[1]
 	var/datum/integrated_io/O = outputs[1]
 	O.data = null
-	var/turf/T = get_turf(src)
-	var/list/nearby_things = view(radius,T)
-	var/list/valid_things = list()
 	var/list/input_list = list()
 	input_list = I.data
-	for(var/item in input_list)
-		if(!isnull(item) && !isnum(item))
-			if(istext(item))
-				for(var/atom/thing in nearby_things)
-					if(findtext(addtext(thing.name," ",thing.desc), item, 1, 0) )
+	if(length(input_list))	//if there is no input don't do anything.
+		var/turf/T = get_turf(src)
+		var/list/nearby_things = view(radius,T)
+		var/list/valid_things = list()
+		for(var/thing in nearby_things)
+			var/atom/AM = thing
+			if(!CIRCUIT_VALID_ATOM(AM))
```

Comment:
oh god why you iterating huge list twice?why don't you move this check somewhere in main cycle?

Reply:
your main cycle is a view() you can't override that.

Reply:
oh you already iterate everything lmao